rosa_tutorials/rosa-mobb-aws-load-balancer-operator.adoc

```diff
@@ -0,0 +1,345 @@
+$ export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+$ export SCRATCH="/tmp/${ROSA_CLUSTER_NAME}/alb-operator"
+mkdir -p ${SCRATCH}
+echo "Cluster: ${ROSA_CLUSTER_NAME}, Region: ${REGION}, OIDC Endpoint: ${OIDC_ENDPOINT}, AWS Account ID: ${AWS_ACCOUNT_ID}"
+----
+
+== AWS VPC and subnets
+
+[NOTE]
+====
+This section only applies to BYO VPC clusters, if you let ROSA create your VPCs you can skip to the following Installation section. You can skip this section if you already installed xref:../rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc#rosa-sts-creating-a-cluster-quickly[Aa Multi-AZ ROSA Classic cluster]. 
+====
+
+.Click to expand the procedure to set up AWS VPC and subnets
+[%collapsible]
+====
+. Set Variables describing your VPC and Subnets:
++
+[source,terminal]
+----
+$ export VPC_ID=<vpc-id>
+$ export PUBLIC_SUBNET_IDS=<public-subnets>
+$ export PRIVATE_SUBNET_IDS=<private-subnets>
+$ export CLUSTER_NAME=$(oc get infrastructure cluster -o=jsonpath="{.status.infrastructureName}")
+----
++
+. Tag VPC with the cluster name: 
++
+[source,terminal]
+----
+$ aws ec2 create-tags --resources ${VPC_ID} --tags Key=kubernetes.io/cluster/${CLUSTER_NAME},Value=owned --region ${REGION}
+----
++
+. Add tags to Public Subnets: 
++
+[source,terminal]
+----   
+$ aws ec2 create-tags \
+     --resources ${PUBLIC_SUBNET_IDS} \
+     --tags Key=kubernetes.io/role/elb,Value='' \
+     --region ${REGION}
+----
++
+. Add tags to Private Subnets: 
++
+[source,terminal]
+----
+$ aws ec2 create-tags \
+     --resources "${PRIVATE_SUBNET_IDS}" \
+     --tags Key=kubernetes.io/role/internal-elb,Value='' \
+     --region ${REGION}
+----
+====
+
+== Installation
+
+. Create Policy for the ALB Controller: 
++ 
+[NOTE]
+====
+Policy is from link:https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/v2.4.4/docs/install/iam_policy.json[ALB controller policy] plus subnet create tags permission. This is required by the operator.
+==== 
++
+[source,terminal]
+----
+$ oc new-project aws-load-balancer-operator
+  POLICY_ARN=$(aws iam list-policies --query \
+     "Policies[?PolicyName=='aws-load-balancer-operator-policy'].{ARN:Arn}" \
+     --output text)
+   if [[ -z "${POLICY_ARN}" ]]; then
+     wget -O "${SCRATCH}/load-balancer-operator-policy.json" \
+       https://raw.githubusercontent.com/rh-mobb/documentation/main/content/docs/rosa/aws-load-balancer-operator/load-balancer-operator-policy.json
+     POLICY_ARN=$(aws --region "$REGION" --query Policy.Arn \
+     --output text iam create-policy \
+     --policy-name aws-load-balancer-operator-policy \
+     --policy-document "file://${SCRATCH}/load-balancer-operator-policy.json")
+   fi
+   echo $POLICY_ARN
+----
++
+. Create trust policy for ALB Operator: 
++
+[source,terminal]
+----   
+$ cat <<EOF > "${SCRATCH}/trust-policy.json"
```

Comment:
it complains because there's spaces preceeding the json.  this should fix it.

```
cat <<EOF > "${SCRATCH}/trust-policy.json"
{
 "Version": "2012-10-17",
 "Statement": [
 {
 "Effect": "Allow",
 "Condition": {
   "StringEquals" : {
     "${OIDC_ENDPOINT}:sub": ["system:serviceaccount:aws-load-balancer-operator:aws-load-balancer-operator-controller-manager", "system:serviceaccount:aws-load-balancer-operator:aws-load-balancer-controller-cluster"]
   }
 },
 "Principal": {
   "Federated": "arn:aws:iam::$AWS_ACCOUNT_ID:oidc-provider/${OIDC_ENDPOINT}"
 },
 "Action": "sts:AssumeRoleWithWebIdentity"
 }
 ]
}
EOF
```

Reply:
Done